src/main/duke/task/Deadline.java

```diff
@@ -0,0 +1,46 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    private LocalDate time;
+
+    /**
+     * Constructor for a Deadline object.
+     * @param name the name for this Deadline task
+     * @param time the time for the deadline
+     */
+    public Deadline(String name, LocalDate time) {
+        super(name);
+        this.time = time;
+    }
+
+    /**
+     * Constructor for a Deadline object with the ability to set completion status.
+     * @param name the name for this Deadline task
+     * @param time the time for the deadline
+     * @param completed the completion status of this Deadline.
+     */
+    public Deadline(String name, LocalDate time, String completed) {
+        super(name, completed);
+        this.time = time;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("[D] %s (by: %s)", super.toString(),
+                time.format(DateTimeFormatter.ofPattern("MMM d yyyy")));
+    }
+
+    @Override
+    public String[] toArray() {

```

Comment:
This seems to behave differently that the typical toArray() where an ADT(abstract data type) is converted to Array.
Maybe you can name it as toSaveFormat() ?

Reply:
Good point. I've renamed it to `toSaveFormatArray` for now.